src/screens/transactions/TransactionDetailsScreen.tsx

```diff
@@ -20,45 +20,79 @@ import {
   formatValueForDisplay,
 } from '../../utils/displayUtils';
 import {
-  LeftAlignedContainer,
+  CenteredOneLine,
+  LeftAlignContainerWithMargins,
   MediumText,
   Size14BoldText,
   Title,
 } from './styles';
+import {
+  SortVoucherOption,
+  useSortVoucherReducer,
+} from '../../utils/transactionUtils';
+import SortModal from '../../components/transactions/SortModal';
+import SortAndFilterButton from '../../components/transactions/SortAndFilterButton';
+import { SortAndFilterDummy } from '../../components/transactions/styles';
 
 export default function TransactionDetailsScreen({
   route,
   navigation,
 }: TransactionStackScreenProps<'TransactionDetailsScreen'>) {
   const { transactionUuid } = route.params;
   const [transactionData, setTransactionData] = useState<Transaction>();
-  const [voucherArray, setVoucherArray] = useState<Voucher[]>();
+  const [defaultVoucherArray, setDefaultVoucherArray] = useState<Voucher[]>([]);
+  const [displayedVoucherArray, setDisplayedVoucherArray] = useState<Voucher[]>(
+    [],
+  );
   const [isRefreshing, setIsRefreshing] = useState(false);
 
-  const onRefresh = React.useCallback(() => {
-    setIsRefreshing(true);
-    setTimeout(() => {
-      setIsRefreshing(false);
-    }, 1000);
-  }, []);
+  const [sortModalIsVisible, setSortModalIsVisible] = useState(false);
+  const { sortVoucherState, sortVoucherDispatch } = useSortVoucherReducer(
+    displayedVoucherArray,
+    defaultVoucherArray,
+    setDisplayedVoucherArray,
+  );
 
-  useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const data = await getTransaction(transactionUuid);
+  const sortButtonText = ['SN', 'SN', 'Date', 'Date'];
+  const sortDescriptionText = [
+    'Serial Number: High to Low',
+    'Serial Number: Low to High',
+    'Date: Newest',
```

Comment:
I tried sorting by 'Date: Newest' in the app and it doesn't work? Invoices are just randomly sorted.

Reply:
Noticed similar issue